stdlib/map.gr

```diff
@@ -51,24 +101,29 @@ export let set = (key, value, map) => {
   let bucket = buckets[idx];
   match (bucket) {
     | None => {
-      buckets[idx] := Some({ key: box(key), value: box(value), next: None });
+      buckets[idx] := Some({ key: box(key), value: box(value), next: box(None) });
       map.size += 1;
     }
-    | Some(cell) => {
-      if (replaceInBucket(key, value, cell)) {
-        buckets[idx] := Some({ key: box(key), value: box(value), next: bucket });
+    | Some(node) => {
+      if (replaceInBucket(key, value, node)) {
+        buckets[idx] := Some({ key: box(key), value: box(value), next: box(bucket) });
         map.size += 1;
       };
     }
-  };
-  # TODO: Need to check and possibly resize the buckets array
+  }
+  # Resize if there are more than 2x the amount of nodes as buckets
```

Comment:
Oh this is interesting. I figured resizing would happen sooner!

Reply:
Hah, it was just taken from BuckleScript but they do `lsl` so I added a comment to explain what I learned 😛